### PR TITLE
NAS-126988 / 24.10 / Fix edge case in container pruning test

### PR DESCRIPTION
**Pull request by Qubad786**
## Problem
In the test `test_pruning_for_deleted_chart_release_images`, an assertion error still occurs sometimes even after waiting for a container to stop using its ID. After investigation, it appears that if a container of a different ID from the same image is still running, it can also cause this test to fail. This can happen if the container is still running from a previous test.

## Solution
Wait for the container to terminate in the previous test `test_pruning_for_deleted_chart_release_images`.

### CODE REVIEW

**Review by yocalebo — APPROVED** [review]

File: tests/api2/test_024_container.py [hunk 1]
```diff
@@ -49,6 +49,19 @@ def chart_release(app_name, values):
...
+    def wait_for_containerd(container_id):
+        timeout = 60
+        while True:
+            if container_id not in [container['id'] for container in call('docker.container.query')]:
+                break
+
+            time.sleep(10)
+            timeout -= 10
+            if timeout <= 0:
+                fail(f'{container_id!r} container was not deleted in 60 seconds')
```
Comment (yocalebo) [inline tests/api2/test_024_container.py:62]: Wait, we probably don't want to call `fail()` here since that will exit the _entire_ CI run. We just need to mark all tests that depend on this container being deleted to be skipped.

**Review by yocalebo — CHANGES_REQUESTED** [review]

File: tests/api2/test_024_container.py (hunk 1 above)
Comment (Qubad786) [reply tests/api2/test_024_container.py:62]: @yocalebo it won't fail the entire CI run and just the test in question https://github.com/truenas/middleware/blob/472ca363161b0c2c7d14065847d567027933e613/src/middlewared/middlewared/test/integration/utils/pytest.py#L6

Comment (themylogin) [reply tests/api2/test_024_container.py:62]: @Qubad786 it will fail the whole test suite exactly as the comment says

Comment (Qubad786) [reply tests/api2/test_024_container.py:62]: This was already being used and i assumed it was to fail the test in question..apologies did not go through the comment.
Fixing it and it's usages ✅

File: tests/api2/test_023_kubernetes.py [hunk 2]
```diff
@@ -82,7 +82,7 @@ def test_07_kubernetes_pods_stats(request):
                     break
 
             if timeout <= 0:
-                fail('Time to setup kubernetes exceeded 150 seconds')
+                pytest.fail('Time to setup kubernetes exceeded 150 seconds')
```
Comment (themylogin) [inline tests/api2/test_023_kubernetes.py:85]: `pytest.fail` should not be used in our code as it can lead to effects described in `middlewared.test.integration.utils.fail` comment

We should fail the entire dependency (`setup_kubernetes` in this case) to prevent k3s tests from running

File: tests/api2/test_024_container.py [hunk 3]
```diff
@@ -49,6 +49,19 @@ def chart_release(app_name, values):
...
+    def wait_for_containerd(container_id):
+        timeout = 60
+        while True:
+            if container_id not in [container['id'] for container in call('docker.container.query')]:
+                break
+
+            time.sleep(10)
+            timeout -= 10
+            if timeout <= 0:
+                pytest.fail(f'{container_id!r} container was not deleted in 60 seconds')
```
Comment (themylogin) [inline tests/api2/test_024_container.py:62]: Same here

**Review by themylogin — APPROVED** [review]
Sorry for misunderstanding, if you only want to fail one test then this is the way to go
